app/main/views/reset_password.py

```diff
@@ -139,3 +141,87 @@ def update_password(token):
                                email_address=email_address,
                                form=form,
                                token=token), 400
+
+
+@main.route('/change-password', methods=["GET", "POST"])
+@login_required
+def change_password():
+    form = ChangeOldPasswordForm()
+    dashboard_url = get_user_dashboard_url(current_user)
+
+    if request.method == 'POST':
+        if form.validate_on_submit():
+            # Make sure old password is valid
+            user_json = data_api_client.authenticate_user(
+                current_user.email_address,
+                form.old_password.data)
+            if user_json:
+                response = data_api_client.update_user_password(
+                    current_user.id,
+                    form.password.data,
+                    updater=current_user.email_address
+                )
+                if response:
+                    current_app.logger.info(
+                        "User {user_id} successfully changed their password",
+                        extra={'user_id': current_user.id}
+                    )
+
+                    notify_client = DMNotifyClient(current_app.config['DM_NOTIFY_API_KEY'])
+
+                    token = generate_token(
+                        {
+                            "user": current_user.id
+                        },
+                        current_app.config['SHARED_EMAIL_KEY'],
+                        current_app.config['RESET_PASSWORD_SALT']
+                    )
+
+                    try:
+                        notify_client.send_email(
+                            current_user.email_address,
+                            template_id=current_app.config['NOTIFY_TEMPLATES']['change_password_alert'],
+                            personalisation={
+                                'url': url_for('main.reset_password', token=token, _external=True),
+                            },
+                            reference='change-password-alert-{}'.format(hash_string(current_user.email_address))
+                        )
+                    except EmailError as e:
+                        current_app.logger.error(
+                            "{code}: Password change alert email for email_hash {email_hash} failed to send. "
+                            "Error: {error}",
+                            extra={
+                                'email_hash': hash_string(current_user.email_address),
+                                'error': str(e),
+                                'code': 'login.password-change-alert-email.notify-error'
+                            }
+                        )
+                        abort(503, response="Failed to send password change alert.")
```

Comment:
This `try..except` was copied from the password reset view. I think the idea was that we wanted it to fail noisily for our benefit rather than for the user's, so we know that Notify is having problems? Agreed that it's not very useful for the user to get a 500 though.